acvm/src/pwg/block.rs

```diff
@@ -6,34 +6,33 @@ use acir::{
     FieldElement,
 };
 
-use super::{
-    arithmetic::{ArithmeticSolver, GateStatus},
-    insert_value,
-};
-use super::{OpcodeNotSolvable, OpcodeResolution, OpcodeResolutionError};
+use super::{arithmetic::ArithmeticSolver, get_value, insert_value};
+use super::{OpcodeResolution, OpcodeResolutionError};
+
+type MemoryIndex = u32;
 
 /// Maintains the state for solving Block opcode
 /// block_value is the value of the Block at the solved_operations step
 /// solved_operations is the number of solved elements in the block
 #[derive(Default)]
 pub(super) struct BlockSolver {
-    block_value: HashMap<u32, FieldElement>,
+    block_value: HashMap<MemoryIndex, FieldElement>,
     solved_operations: usize,
 }
 
 impl BlockSolver {
-    fn insert_value(&mut self, index: u32, value: FieldElement) {
+    fn write_memory_index(&mut self, index: MemoryIndex, value: FieldElement) {
         self.block_value.insert(index, value);
     }
 
-    fn get_value(&self, index: u32) -> Option<FieldElement> {
-        self.block_value.get(&index).copied()
+    fn read_memory_index(&self, index: MemoryIndex) -> FieldElement {
+        self.block_value.get(&index).copied().expect("Should not read uninitialized memory")
```

Comment:
I've chosen to retain this as an `expect` for now. We can turn this into a proper error type later.